.gitignore

```diff
@@ -23,3 +23,5 @@
 .#*
 /.cache
 /bin/spackc
+*.in.log
```

Comment:
are these spack specific? If not, maybe remove it from PR

Reply:
These are created when Spack is invoked with `-d` like `spack -d install...`

Reply:
Ah, then it makes sense that I see so many of them, as I always use that flag to assist my learning how things work.

Does it make sense to keep this change which causes git to ignore them in that case?  Should this commit be a separate PR?